pkg/abstract/slices.go

```diff
@@ -26,6 +26,20 @@ func SliceDelete[Elt any](slice *[]Elt, index int) {
 	*slice = (*slice)[:lastIndex]
 }
 
+// SliceFilter makes a new slice that contains the members of the given slice that pass the given filter
+func SliceFilter[Elt any](input []Elt, good func(Elt) bool) []Elt {
```

Comment:
SliceFilter implies to me - in-place filtering , not allocating and creating a new slice.

Reply:
E.g., NewFilteredSlice   or CreateFilteredSlice

Reply:
I pushed a revision giving this method a more verbose name.